src/Autocomplete/src/AutocompleteResultsExecutor.php

```diff
@@ -46,14 +50,57 @@ public function fetchResults(EntityAutocompleterInterface $autocompleter, string
 
         $entities = $queryBuilder->getQuery()->execute();
 
-        $results = [];
+        $options = [];
+        $optgroupLabels = [];
+
+        if (null !== $groupBy = $autocompleter->getFormOption('group_by')) {
+            if (\is_string($groupBy)) {
+                $groupBy = new PropertyPath($groupBy);
+            }
+
+            if ($groupBy instanceof PropertyPathInterface) {
+                $accessor = PropertyAccess::createPropertyAccessor();
```

Comment:
Can we just inject the `property_path` service into this class instead of creating the new object?